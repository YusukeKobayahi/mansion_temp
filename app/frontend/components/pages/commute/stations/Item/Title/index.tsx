import styles from "./index.module.scss";

type TitleProps = {
  children: React.ReactText;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
};

const Title: React.FC<TitleProps> = ({ children, onClick }: TitleProps) => {
  return (
    <>
      <button className={styles.title} onClick={onClick}>
        <h3>{children}</h3>
      </button>
    </>
  );
};

export default Title;
