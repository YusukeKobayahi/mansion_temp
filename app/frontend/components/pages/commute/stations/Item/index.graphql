fragment MsLibraryRailwayStationToStationItem on RailwayStationToStation {
  minutes
  transferTime
  endStation {
    id
    name
    groupCode
    mansionCount
    sameStations {
      line {
        id
        name
      }
    }
    averageMansionPrice {
      averageSquarePrice
    }
    city {
      name
      prefecture {
        name
      }
    }
  }
}
