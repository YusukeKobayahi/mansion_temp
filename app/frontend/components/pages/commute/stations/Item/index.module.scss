@import "~/styles/variables";
@import "~/styles/utils";

.layout {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  overflow: hidden;
  margin-bottom: 24px;

  @include screenMd {
    background-color: #fcfcfc;
    border-radius: 4px;
    border: 1px solid $color-main;
  }
}

.checkbox {
  grid-column: 1 / 2;
  grid-row: 1 / 2;
  margin-top: 10px;

  @include screenMd {
    margin-top: 0px;
    padding: 10px;
    border-right: 1px solid $color-main;
  }
}

.title {
  grid-column: 2/ 13;
  grid-row: 1 / 2;
}

.content {
  grid-column: 2 / 13;
  grid-row: 2 / 3;
  margin-bottom: 15px;

  .body {
    width: 100%;
    margin: 0 auto;
    padding: 20px;
    color: #646464;

    .pcTable {
      width: 100%;
      font-size: 13px;
      display: table;
      table-layout: fixed;

      td {
        color: $color-text-main;
        padding: 15px;
        border: solid;
        border-right: 1px;
        border-left: 1px;
        border-color: #e5e5e5;
      }

      tr td:nth-child(odd) {
        background-color: #f9f9f9;
      }
    }
    .spTable {
      width: 100%;
      display: table;
      table-layout: fixed;
      margin-bottom: 20px;

      tr {
        width: 100%;
        td {
          color: $color-text-main;
          padding: 2px;
          font-weight: lighter;
        }
      }
    }

    .heading {
      margin-bottom: 24px;
    }
  }

  @include screenMd {
    border-top: 1px solid $color-main;
    grid-column: 1 / 13;
  }
}

.subText {
  color: #9d9d9d;
  font-size: 12px;
  line-height: 16px;
}
