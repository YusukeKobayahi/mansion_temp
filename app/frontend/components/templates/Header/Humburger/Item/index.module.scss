@import "~/styles/variables";
@import "~/styles/utils";

.title {
  display: flex;
  align-items: center;
  width: 100%;
  color: $color-main;
  padding: 13px 20px;
}

.arrow {
  margin-left: auto;
  margin-right: 7px;
  background-color: transparent;
  color: $color-main;
  span {
    display: block;
    width: 11px;
    height: 11px;
    border-right: 1px solid #746742;
    border-bottom: 1px solid #746742;
    transform-origin: center center;
    transform: rotate(-45deg);
  }
}

/**
 * Background color
 */
.bg-white {
  background-color: white;
}
.bg-transparent {
  background-color: transparent;
}

/**
 * FontSize
 */
.font-size-sm {
  font-size: 14px;
  font-weight: normal;
  line-height: 1.5;
}
.font-size-md {
  font-size: 16px;
  font-weight: bold;
  line-height: 1.5;
}

/**
 * UnderLine
 */

.underline-solid {
  border-bottom: 1px solid #d8d8d8;
}
.underline-dashed {
  border-bottom: 1px dashed #d8d8d8;
}
.underline-none {
  border-bottom: 1px none #d8d8d8;
}
