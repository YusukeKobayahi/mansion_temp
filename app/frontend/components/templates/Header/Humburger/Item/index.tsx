import styles from "./index.module.scss";
import classnames from "classnames";
import ConditionalWrapper from "~/components/commons/ConditionalWrapper";
import React from "react";

type ItemProps = {
  text: React.ReactNode;
  underline?: "solid" | "dashed" | "none";
  fontSize?: "sm" | "md";
  arrow?: boolean;
  type?: "button" | "normal";
  bgColor?: "white" | "transparent";
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
};

// eslint-disable-next-line react/display-name
const Item = React.forwardRef(
  (
    {
      text,
      onClick,
      underline = "none",
      fontSize = "md",
      arrow = true,
      type = "button",
      bgColor = "white",
    }: ItemProps,
    ref: React.Ref<HTMLButtonElement>
  ) => {
    const titleClasses = classnames(
      [styles.title],
      [styles[`bg-${bgColor}`]],
      [styles[`underline-${underline}`]],
      [styles[`font-size-${fontSize}`]]
    );

    const wrappers = {
      // eslint-disable-next-line react/display-name
      normal: (children: React.ReactNode) => {
        return <div className={titleClasses}>{children}</div>;
      },
      // eslint-disable-next-line react/display-name
      button: (children: React.ReactNode) => {
        return (
          <button ref={ref} className={titleClasses} onClick={onClick}>
            {children}
          </button>
        );
      },
    };
    return (
      <ConditionalWrapper wrapper={wrappers[type]}>
        <div>{text}</div>
        <div
          className={classnames(styles.arrow, arrow ? "visible" : "invisible")}
        >
          <span></span>
        </div>
      </ConditionalWrapper>
    );
  }
);

export default Item;
