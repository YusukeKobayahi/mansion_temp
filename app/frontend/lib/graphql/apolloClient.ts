// 参考
// https://github.com/vercel/next.js/tree/canary/examples/with-apollo

import { useMemo } from "react";
import { ApolloClient, ApolloLink, InMemoryCache } from "@apollo/client";
import { createHttpLink } from "apollo-link-http";
import { from } from "apollo-link";
import { createFragmentArgumentLink } from "apollo-link-fragment-argument";
import { mergeDeep, relayStylePagination } from "@apollo/client/utilities";
import { isSSR } from "~/lib/utils";

/* eslint-disable @typescript-eslint/no-explicit-any -- キャッシュの型を事前に決められないため */
type CacheState = Record<string, any> | null;
/* eslint-enable @typescript-eslint/no-explicit-any */

/* eslint-disable @typescript-eslint/naming-convention -- GraphQLのタイプ名 (このconventionに従わない)を指定したいので */
const createCache = () =>
  new InMemoryCache({
    typePolicies: {
      Mansion: {
        keyFields: ["uniqueCode"],
        fields: {
          salesHistories: relayStylePagination(),
        },
      },
    },
  });
/* eslint-enable @typescript-eslint/naming-convention */

const link = from([
  createFragmentArgumentLink(),
  createHttpLink({
    uri: `${process.env.NEXT_PUBLIC_CORE_API_URL}/consumer/graphql`,
    fetchOptions: {
      mode: "cors",
    },
  }),
]);

function createApolloClient() {
  return new ApolloClient({
    ssrMode: isSSR(),
    // See: https://github.com/apollographql/apollo-link/issues/1258#issuecomment-604031488
    // TODO: @apollo/clientバージョン上げたら直るかも
    link: (link as unknown) as ApolloLink,
    cache: createCache(),
    defaultOptions: {
      watchQuery: {
        // SSR時にuseQueryによってリクエストしてレンダリングする場合、
        // getServerSidePropsからキャッシュが渡っている必要がある。(でないと描画されない)
        // そのため、SSR時は常にcacheから取得されるべきで、実装意図を明確にするためPolicyも設定しておく
        // (エラーになるわけではない)
        fetchPolicy: isSSR() ? "cache-only" : "cache-first",
      },
    },
  });
}

let apolloClient: ReturnType<typeof createApolloClient>;

export function initializeApollo(
  initialState: CacheState = null
): ApolloClient<CacheState> {
  const _apolloClient = apolloClient ?? createApolloClient();

  // If your page has Next.js data fetching methods that use Apollo Client, the initial state
  // gets hydrated here
  if (initialState) {
    // Get existing cache, loaded during client side data fetching
    const existingCache = _apolloClient.extract();

    // Merge the existing cache into data passed from getStaticProps/getServerSideProps
    const data = mergeDeep(initialState, existingCache);

    // Restore the cache with the merged data
    _apolloClient.cache.restore(data);
  }
  // For SSG and SSR always create a new Apollo Client
  if (isSSR()) return _apolloClient;
  // Create the Apollo Client once in the client
  if (!apolloClient) apolloClient = _apolloClient;

  return _apolloClient;
}

export function useApollo(initialState: CacheState): ApolloClient<CacheState> {
  const store = useMemo(() => initializeApollo(initialState), [initialState]);
  return store;
}
