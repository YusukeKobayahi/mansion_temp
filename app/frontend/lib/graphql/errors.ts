export interface GraphQLError {
  graphQLErrors?: {
    message: string;
    extensions?: {
      code: string;
    };
  }[];
}

export const isNotFoundError = (err: GraphQLError): boolean => {
  const graphQLErrors = err.graphQLErrors;
  if (graphQLErrors == null) return false;

  return graphQLErrors[0]?.extensions?.code === "NOT_FOUND";
};
