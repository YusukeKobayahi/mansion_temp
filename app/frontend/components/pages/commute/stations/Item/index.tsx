import { xor } from "lodash";
import { yenFormat, buildLibraryPagePath } from "~/lib/utils";
import { ReactSetStateType } from "~/lib/types";
import {
  MsLibraryRailwayStationToStationItemFragment,
  CommuteStationsQuery,
} from "~/graphql/generated";

import CheckBox from "~/components/commons/CheckBox";
import Title from "~/components/pages/commute/stations/Item/Title";

import styles from "./index.module.scss";
import classnames from "classnames";
import Router from "next/router";

type ItemProps = {
  data: MsLibraryRailwayStationToStationItemFragment;
  st: string[];
  setSt: ReactSetStateType<string[]>;
  fromStation: CommuteStationsQuery["station"];
};

const Item: React.FC<ItemProps> = ({
  data,
  st,
  setSt,
  fromStation,
}: ItemProps) => {
  const { minutes, endStation: toStation, transferTime } = data;
  const {
    mansionCount = 0,
    sameStations = [],
    averageMansionPrice = { averageSquarePrice: 0 },
    city = { name: "", prefecture: { name: "" } },
  } = toStation ? toStation : {};

  const location = `${city?.prefecture?.name}${city?.name}`;
  const lineInfo = sameStations.map((station) => station?.line?.name).join(",");
  const marketPrice = yenFormat(averageMansionPrice?.averageSquarePrice || 0);
  const toStationPath = buildLibraryPagePath("stations", String(toStation?.id));

  return (
    <div className={styles.layout}>
      <div className={styles.checkbox}>
        <CheckBox
          value={String(toStation?.id)}
          checked={st.includes(String(toStation?.id))}
          onChange={(e) => {
            setSt(xor([...st], [e.target.value]));
          }}
        />
      </div>
      <div className={styles.title}>
        <Title onClick={() => Router.push(toStationPath)}>
          {`${toStation?.name}駅` || ""}
        </Title>
      </div>
      <div className={styles.content}>
        <div className={styles.body}>
          <table className={classnames(styles.pcTable, "pcOnly")}>
            <tbody>
              <tr>
                <td>{fromStation?.name}駅までの時間</td>
                <td>{minutes}分</td>

                <td>乗り換え回数</td>
                <td>{transferTime}回</td>
              </tr>
              <tr>
                <td>平均相場(㎡)</td>
                <td> {marketPrice}</td>

                <td>物件数</td>
                <td> {mansionCount}件</td>
              </tr>
              <tr>
                <td>所在地</td>
                <td colSpan={3}>{location}</td>
              </tr>
              <tr>
                <td>路線情報</td>
                <td colSpan={3}> {lineInfo}</td>
              </tr>
            </tbody>
          </table>
          <div className="spOnly">
            <table className={styles.spTable}>
              <tbody>
                <tr>
                  <td>{fromStation?.name}駅まで</td>
                  <td>
                    {minutes}分(乗り換え{transferTime}回)
                  </td>
                </tr>
                <tr>
                  <td>平均相場(㎡)</td>
                  <td> {marketPrice}</td>
                </tr>
                <tr>
                  <td>物件数</td>
                  <td> {mansionCount}件</td>
                </tr>
              </tbody>
            </table>
          </div>
          <div className={classnames(styles.subText, "spOnly")}>
            <div>{location}</div>
            <div>{lineInfo}</div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Item;
