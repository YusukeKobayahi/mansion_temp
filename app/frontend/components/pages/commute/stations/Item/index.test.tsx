import { render } from "@testing-library/react";
import mockData from "../fixtures/mockData.json";
import { CommuteStationsQuery } from "~/graphql/generated";
import Item from ".";

describe("components/pages/commute/stations/Item", () => {
  const mock = (mockData as unknown) as CommuteStationsQuery;

  it("正常に表示される", () => {
    const station = mock.station;
    const stationToStation = station.stationToStations?.[0];
    const { asFragment } = render(
      <Item
        data={stationToStation!}
        st={["1321", "7655"]}
        setSt={() => {
          return;
        }}
        fromStation={station}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
