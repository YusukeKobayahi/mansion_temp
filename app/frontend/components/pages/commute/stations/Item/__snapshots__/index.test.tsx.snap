// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/commute/stations/Item 正常に表示される 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <div
      class="checkbox"
    >
      <div
        class="checkbox"
      >
        <input
          id="checkbox-undefined"
          type="checkbox"
          value="undefined"
        />
        <label
          class=""
          for="checkbox-undefined"
        >
          ✔︎
        </label>
      </div>
    </div>
    <div
      class="title"
    >
      <button
        class="title"
      >
        <h3>
          undefined駅
        </h3>
      </button>
    </div>
    <div
      class="content"
    >
      <div
        class="body"
      >
        <table
          class="pcTable pcOnly"
        >
          <tbody>
            <tr>
              <td>
                品川駅までの時間
              </td>
              <td>
                14分
              </td>
              <td>
                乗り換え回数
              </td>
              <td>
                0回
              </td>
            </tr>
            <tr>
              <td>
                平均相場(㎡)
              </td>
              <td>
                 円
              </td>
              <td>
                物件数
              </td>
              <td>
                 0件
              </td>
            </tr>
            <tr>
              <td>
                所在地
              </td>
              <td
                colspan="3"
              />
            </tr>
            <tr>
              <td>
                路線情報
              </td>
              <td
                colspan="3"
              >
                 
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="spOnly"
        >
          <table
            class="spTable"
          >
            <tbody>
              <tr>
                <td>
                  品川駅まで
                </td>
                <td>
                  14分(乗り換え0回)
                </td>
              </tr>
              <tr>
                <td>
                  平均相場(㎡)
                </td>
                <td>
                   円
                </td>
              </tr>
              <tr>
                <td>
                  物件数
                </td>
                <td>
                   0件
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="subText spOnly"
        >
          <div />
          <div />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
